shell/platform/android/io/flutter/embedding/android/FlutterFragment.java

```diff
@@ -486,9 +467,34 @@ protected String getDartEntrypointFunctionName() {
     return FlutterView.TransparencyMode.valueOf(transparencyModeName);
   }
 
+  @Override
+  public void onStart() {
```

Comment:
How often do we want this to repeat? `onStart` isn't executed every time the activity is brought into the foreground either unfortunately. `onResume` would be the right fit for that case.

On the flip side, is `flutterView.detachFromFlutterEngine();` the mirror of this? If so it should probably be moved into `onStop` so it's called in equivalent situations, or `onPause` if this ultimately needs to be moved into `onResume`.

Reply:
Placing this in onResume() would imply that the UI could attach/detach while the Activity is still visible, but developers shouldn't do that. For example, when a dialog pops up, onPause() is invoked. I think the only reason to put this in onResume() would be if we expected to detach in onPause(), but in that case the Flutter UI would disappear out from under the dialog. So I do think onStart() is the correct lifecycle hook.

As for detaching, the reason that we need to attach in onStart() is in case the underlying engine was re-used in another Activity. So when returning to this Activity we re-attach in onStart(). Then there is detaching, and you asked if detaching should be symmetric to attaching. Technically we could detach in onStop() as you suggested, but it also occurred to me that in most cases that detachment is superfluous. In most cases an engine is probably not being re-used, but we'd go through detachment every time the FlutterActivity hits onStop(). 

I don't know if there is a meaningful performance risk to detaching like that, but I figured we can avoid it by not explicitly detaching until we know the FlutterFragment is detaching from the Activity or being destroyed.

Does that make sense? Do you still think detachment should happen in on onStop() to be symmetric?

Reply:
The `onStart()` vs `onResume()` change makes sense to me.

For detaching, I guess my remaining question is is there a consequence for the engine not being detached in line with another potential attach() call? Is that a potential issue or is there no larger issue there? If it doesn't matter then it's fine to be asymmetrical, but I was just assuming that there would be some kind of consequence for overlap there so you'd want to make sure it was torn up and down more predictably.

Reply:
Looking again at attachment, given that a number of things are setup there, and we may discover that they need to be cleaned up in detachment, I do think the safer thing is to be symmetric. So I'm going to move detachment to `onStop()`.